pkg/controller/deployment/deployment_controller.go

```diff
@@ -166,14 +172,14 @@ func (dc *DeploymentController) Run(workers int, stopCh <-chan struct{}) {
 
 func (dc *DeploymentController) addDeployment(obj interface{}) {
 	d := obj.(*apps.Deployment)
-	klog.V(4).Infof("Adding deployment %s", d.Name)
+	klog.V(4).InfoS("Adding deployment", klogKeyDeployment, klog.KObj(d))
```

Comment:
i see repetition for deployment in `"Adding deployment" deployment="kube-system/coredns"`

This can simply be `"Adding" deployment="kube-system/coredns"`

Reply:
Using complete sentences in log message is in line with [Structured Logging migration instructions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md) and [basic examples in the klog package](https://github.com/kubernetes/klog/blob/master/klog.go#L1409).
```
// >> klog.InfoS("Pod status updated", "pod", "kubedns", "status", "ready")
// output:
// >> I1025 00:15:15.525108       1 controller_utils.go:116] "Pod status updated" pod="kubedns" status="ready"
```

That repetition is accidental. It occurs because I use `deployment` as a keyword.
If I used something else instead of `deployment` we would not see any repetition here. For example:
```
klog.V(4).InfoS("Adding deployment", "id", klog.KObj(d))
// "Adding deployment" id="kube-system/coredns"
```

On the opposite side reducing the log message to `Adding` would lead to an incomplete message which wouldn't explain what exactly we are adding. For example:
```
klog.V(4).InfoS("Adding", "id", klog.KObj(d))
// "Adding" id="kube-system/coredns"
```

Another disadvantage of using standalone `Adding` is that when there are more than 1 key/value pair provided, it creates ambiguity because it is not clear which of the keywords `Adding` is referring to. It is not guaranteed that downstream libraries preserve keyword ordering. 

```
klog.V(4).InfoS("Adding", "pod", klog.KObj(pod), "deployment", klog.KObj(deployment), "replicaSet", klog.KObj(replicaSet))
// "Adding" pod="kube-system/coredns-xxxx" deployment="kube-system/coredns-xxxx" replicaSet="kube-system/coredns-xxxx"
// or
// "Adding" deployment="kube-system/coredns-xxxx" replicaSet="kube-system/coredns-xxxx" pod="kube-system/coredns-xxxx"
// or
// "Adding" replicaSet="kube-system/coredns-xxxx" deployment="kube-system/coredns-xxxx" pod="kube-system/coredns-xxxx"
```

That's why I believe keeping the subject in the message (i.e. `Adding deployment`) is required.

Let me know your thoughts please.

Reply:
how about we change the key to be simply `name` ? then we will have `"Adding deployment" name="kube-system/coredns"` wdyt ?

Reply:
I like `name`.
However, I believe we should be consistent across different components.
If we use `name` here, we should be using `name` for pod names, statefulSet names, etc.
Otherwise, log readers will be having a hard time trying to figure out which keyword they need to use to get the logs they are interested in.

Let me ping someone from #sig-instrumentation who drives this migration so that they share their vision.
Personally, I don't have a strong opinion here.

Reply:
Hey, author of migration instructions here. Based on current guidelines most correct version would be `"Adding deployment" deployment="kube-system/coredns"`. Some arguments:
* This log will include repetition, but this repetition is required to make meaning of both message and it's metadata independent. Message alone should be able to provide good enough description of event that is happening. 
* We should try to use consistent key names that are depended on context of any specific log. This is with goal of being able to answer queries like "Give me all logs relating to this specific deployment". Meaning of key `name` proposed above is based on context of provided by message which is not structured, so in this log is stores deployment reference, but in next one it could store pod name, making it not very useful for correlating log lines.